lib/template/hil.go

```diff
@@ -0,0 +1,51 @@
+package template
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/hil"
+	"github.com/hashicorp/hil/ast"
+)
+
+// InterpolateHIL processes the string as if it were HIL and interpolates only
+// the provided string->string map as possible variables.
+func InterpolateHIL(s string, vars map[string]string, lowercase bool) (string, error) {
```

Comment:
Do we ever pass `false` for `lowercase`, I couldn't find such a call in this PR.

I generally consider boolean function args as something that should be avoided at all costs. From the caller a literal true/false reads really poorly. Putting the bool into a struct, or replacing it with a polymorphic type makes the calling code a lot easier to read.

In this case a struct is probably unnecessary at this point, but a varargs could work:

```go
func InterpolateHIL(s string, vars map[string]string, valueOp ...func(string) string) { ...}
...
// lowercase 
InterpolateHIL(s, vars, strings.Lower)
```

But if we always use lowercase=true, then maybe we can drop the parameter?

Reply:
AFAICT we don't ever pass false except during tests. While I agree that its unnecessary I was just moving this function to a location where I could use it. We can PR the other change at a later date.